src/run_build.py

```diff
@@ -63,11 +74,11 @@ def main() -> int:
             architecture=args.architecture or manifest.build.architecture,
         )
 
-        build_recorder = BuildRecorder(target)
+        build_recorder = BuildRecorder(target, build_manifest) if args.incremental else BuildRecorder(target)
 
         logging.info(f"Building {manifest.build.name} ({target.architecture}) into {target.output_dir}")
 
-        for component in manifest.components.select(focus=args.components, platform=target.platform):
+        for component in manifest.components.select(focus=components, platform=target.platform):
```

Comment:
Won't this disable building the individual components provided by `--component`?

Reply:
Yes, it would. We could have some enhancements discussed later to accommodate both `--component` and `--incremental`

Reply:
I don't think we should break the existing code compatibilty. All the GH issues created for build failures of any component recommend to build using `--component` args. Example: https://github.com/opensearch-project/alerting/issues/1326#issue-2036863756

In order to unblock and move forward we can do below:
1. Make `--component` and `--incremental` args mutually exclusive
2. Create an issue to accommodate the feature where if `--component <name> --incremental` is passed, it should compare the component from the previous manifest and using the diff make a decision to either rebuilt or skip.

Reply:
Just realize if components list is empty. This will rebuild all components. Looking into this.